### PR TITLE
Added yaml config support

### PR DESCRIPTION
**Pull request by jossjacobo**
Replaced old default.properties

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,45 +14,80 @@ Using maven, include it as a dependency:
...
 </repositories>
 ```
-```
+```xml
 <dependencies>
     <dependency>
         <groupId>com.github.willowtreeapps</groupId>
         <artifactId>conductor</artifactId>
-        <version>2.5.0</version>
+        <version>3.0.0</version>
```
Comment (konwinkler) [inline README.md]: should update the version in the pom too

File: src/main/java/io/ddavison/conductor/DriverUtil.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                    System.err.println("Unknown browser: " + config.getBrowser());
+                    return null;
+            }
+
+            if (!isLocal) {
+                try {
+                    driver = new RemoteWebDriver(config.getHub(), capabilities);
+                } catch (Exception e) {
+                    Log.fatal("Couldn't connect to hub: " + config.getHub().toString());
+                    e.printStackTrace();
```
Comment (konwinkler) [inline src/main/java/io/ddavison/conductor/DriverUtil.java]: Could we print the stack trace through the logger>?

File: src/main/java/io/ddavison/conductor/DriverUtil.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+                }
+            }
+        } catch (Exception x) {
+            Log.fatal("Also see https://github.com/conductor-framework/conductor/wiki/WebDriver-Executables");
```
Comment (konwinkler) [inline src/main/java/io/ddavison/conductor/DriverUtil.java]: Might be a good time to update this url

File: src/main/java/io/ddavison/conductor/Locomotive.java [hunk 4]
```diff
@@ -527,17 +383,17 @@ public Locomotive waitForWindow(String regex) {
...
 
         // when we reach this point, that means no window exists with that title..
-        if (attempts == configuration.retries()) {
-            Assertions.fail("Window with title: " + regex + " did not appear after " + configuration.retries() + " tries. Exiting.");
+        if (attempts == configuration.getRetries()) {
+            Assertions.fail("Window with title: " + regex + " did not appear after " + configuration.getRetries() + " tries. Exiting.");
             return this;
         } else {
-            System.out.println("#waitForWindow() : Window doesn't exist yet. [" + regex + "] Trying again. " + (attempts + 1) + "/" + configuration.retries());
+            System.out.println("#waitForWindow() : Window doesn't exist yet. [" + regex + "] Trying again. " + (attempts + 1) + "/" + configuration.getRetries());
```
Comment (konwinkler) [inline src/main/java/io/ddavison/conductor/Locomotive.java]: Could we use the logger here too?

File: src/test/java/io/ddavison/conductor/FrameworkTest.java [hunk 5]
```diff
@@ -1,28 +1,29 @@
...
 
-@Config(
-        browser = Browser.CHROME,
-        url = "http://ddavison.io/tests/getting-started-with-selenium.htm")
+@Config(browser = Browser.CHROME)
 public class FrameworkTest extends Locomotive {
 
+    private static final String PWD = System.getProperty("user.dir");
+    private static final String DIR = "src/test/resources";
+    static final String TEST_HTML_FILE = String.format("file://%s/%s/selenium-test/selenium.html", PWD, DIR);
```
Comment (konwinkler) [inline src/test/java/io/ddavison/conductor/FrameworkTest.java]: Nice way to get away from the hosted page!

**Comment by jossjacobo** [comment]
@konwinkler done!

File: README.md (hunk 1 above)
Comment (jossjacobo) [reply README.md]: Done.

**Comment by jossjacobo** [comment]
Anything other feedback? @fuzzybinary would you _kindly_ review my PR? ;)

**Pull request merged**